consensus/src/dag/dag_driver.rs

```diff
@@ -139,14 +139,31 @@ impl DagDriver {
 
     pub async fn enter_new_round(&mut self, new_round: Round, strong_links: Vec<NodeCertificate>) {
         debug!("entering new round {}", new_round);
-        // TODO: support pulling payload
+        let payload_filter = {
+            let dag_reader = self.dag.read();
+            let highest_commit_round = dag_reader.highest_committed_anchor_round();
+            if strong_links.is_empty() {
+                PayloadFilter::Empty
+            } else {
+                PayloadFilter::from(
+                    &dag_reader
+                        .reachable(
+                            strong_links.iter().map(|node| node.metadata()),
```

Comment:
I think we need to turn off QS proof broadcast step, otherwise same round might have duplicates.

Reply:
yeah we should make that configurable, but probably not in this pr